frontend/src/app/main/data/workspace/guides.cljs

```diff
@@ -0,0 +1,101 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) UXBOX Labs SL
+
+(ns app.main.data.workspace.guides
+  (:require
+   [app.common.geom.point :as gpt]
+   [app.common.geom.shapes :as gsh]
+   [app.common.spec :as us]
+   [app.common.types.page-options :as tpo]
+   [app.main.data.workspace.changes :as dwc]
+   [app.main.data.workspace.state-helpers :as wsh]
+   [beicon.core :as rx]
+   [cljs.spec.alpha :as s]
+   [potok.core :as ptk]))
+
+(defn make-update-guide [guide]
+  (fn [other]
+    (cond-> other
+      (= (:id other) (:id guide))
+      (merge guide))))
+
+(defn update-guides [guide]
+  (us/verify ::tpo/guide guide)
+  (ptk/reify ::update-guides
+    ptk/WatchEvent
+    (watch [it state _]
+      (let [page-id (:current-page-id state)
+            guides (-> state wsh/lookup-page-options (:guides {}))
+
+            new-guides (assoc guides (:id guide) guide)
+            
+            rch [{:type :set-option
```

Comment:
maybe we can do this using the same approach as in `app.common.pages.changes-builder` ?

Reply:
OK